sdks/python/apache_beam/io/filesystems.py

```diff
@@ -17,24 +17,47 @@
 
 """FileSystems interface class for accessing the correct filesystem"""
 
+import re
+
 from apache_beam.io.filesystem import BeamIOError
 from apache_beam.io.filesystem import CompressionTypes
-from apache_beam.io.filesystems_util import get_filesystem
+from apache_beam.io.filesystem import FileSystem
+
+# pylint: disable=wrong-import-position, unused-import
+from apache_beam.io.localfilesystem import LocalFileSystem
+
+try:
+  from apache_beam.io.gcp.gcsfilesystem import GCSFileSystem
+except ImportError:
+  pass
+# pylint: enable=wrong-import-position, unused-import
 
 
 class FileSystems(object):
   """A class that defines the functions that can be performed on a filesystem.
   All methods are static and access the underlying registered filesystems.
   """
+  URI_SCHEMA_PATTERN = re.compile('(?P<scheme>[a-zA-Z][-a-zA-Z0-9+.]*)://.*')
+
+  @staticmethod
+  def get_scheme(path):
+    match_result = FileSystems.URI_SCHEMA_PATTERN.match(path)
+    if match_result is None:
+      return None
+    return match_result.groupdict()['scheme']
 
   @staticmethod
   def get_filesystem(path):
     """Get the correct filesystem for the specified path
     """
     try:
-      return get_filesystem(path)
+      path_scheme = FileSystems.get_scheme(path)
+      for fs in FileSystem.get_all_subclasses():
```

Comment:
I think we should raise an exception to prevent surprises in case somebody add a file-system with an existing prefix.